source/slang/slang-ast-dump.cpp

```diff
@@ -0,0 +1,615 @@
+// slang-ast-dump.cpp
+#include "slang-ast-dump.h"
+#include <assert.h>
+
+#include "slang-compiler.h"
+
+#include "../core/slang-string.h"
+
+#include "slang-ast-generated-macro.h"
+
+// There could be an argument for using the SourceWriter, but whilst that has some useful stuff
+// like the automatic indentation, it is a little clumsy in some other ways, like how to build up
+// a result via << operators is not available.
```

Comment:
Interesting point. It might be good to create a unified abstraction for outputting formatted text (especially around things like indention handling), and then put all the necessary convenience features into it.